docs/migration-guide.md

```diff
@@ -0,0 +1,135 @@
+We roll out a brand new ckb-java-sdk after a refactor work. The new sdk brings plenty of BREAKING CHANGES incompatible with the deprecated sdk `v1.0.*` and earlier releases.
+
+The breaking changes include
+
+- Type or name change of quite a few fields in RPC type representation.
+- Unified address representation and operation.
+- Transaction signing mechanism by `ScriptGroup`, `ScriptSigner`, and `TransactionSigner`.
+- `TransactionBuilder` and `ScriptHandler` for transaction construction by manual.
+- Operation support for most common scripts.
+- Clean some utils classes and unused classes.
+
+Other underlying breaking changes that a possibly transparent to you
+
+- Molecule serialization in a code-generated way.
+- Unify type adapter for Gson serialization.
+- More robust test.
+
+In the following part, we list the aspects you need to care about most if you are ready to migrate from deprecated sdk to the new one.
+
+## Address
+
+The deprecated sdk uses several classes to represent, generate or process ckb address, including `AddressUtils`, `AddressGenerator`, `AddressParser`, `AddressTools`, and so on. The new sdk unifies all of them and represents ckb address by a single class `Address`.
+
+Parse / decode address
+
+```java
+String encoded = "ckt1qzda..."
+// Before:
+AddressParseResult result = AddressParser.parse(encoded);
+Script script = result.script;
+Network network = result.network;
+
+// Now:
+Address address = Address.decode(encoded);
+Script script = address.getScript();
+Network network = address.getNetwork();
+```
+
+Encode address from script and network
+
+```java
+// Before:
+String encoded = AddressGenerator.generate(network, script);
+
+// Now:
+Address address = new Address(script, network);
+String encoded = address.encode();
+```
+
+## data representation
+
+### hash
+
+The 32-byte hash is widely used in ckb world. The deprecated sdk uses `String` to represent the hash, while the new sdk represents it by `byte[]`.
+
+```java
+// Before:
+String hash = tx.computeHash();
+
+// Now:
+byte[] hash = tx.computeHash(); 
+```
+
+Type should be changed like this in all places where the hash is used.
+
+### Number
+
+To align with [CKB RPC Types](https://github.com/nervosnetwork/ckb/tree/develop/rpc#rpc-types), the new sdk represents `Uint32` by `int`, `Uint64` by `long` and `Uint128` by `BigInteger` for corresponding data. The deprecated sdk does not obey this rule strictly.
+
+Because the number is signed in java but is unsigned in CKB RPC, you should operate the these numeric variables in an unsigned way in sdk.
```

Comment:
```suggestion
Because the number is signed in java but is unsigned in CKB RPC, you should operate these numeric variables in an unsigned way in sdk.
```